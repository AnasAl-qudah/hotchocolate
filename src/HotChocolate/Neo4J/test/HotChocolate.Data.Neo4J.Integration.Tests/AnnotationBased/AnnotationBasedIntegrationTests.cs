using CookieCrumble;
using HotChocolate.Data.Neo4J.Testing;
using HotChocolate.Execution;

namespace HotChocolate.Data.Neo4J.Integration.Tests.AnnotationBased;

[Collection(Neo4JDatabaseCollectionFixture.DefinitionName)]
public class AnnotationBasedIntegrationTests : IClassFixture<Neo4JFixture>
{
    private readonly Neo4JDatabase _database;
    private readonly Neo4JFixture _fixture;

    public AnnotationBasedIntegrationTests(Neo4JDatabase database, Neo4JFixture fixture)
    {
        _database = database;
        _fixture = fixture;
    }

    [Fact]
    public async Task MoviesSchemaIntegrationTests_GetSchema()
    {
        var tester = await _fixture.CreateSchema(_database);
        tester.Schema.MatchSnapshot();
    }

    [Fact]
    public async Task MoviesSchemaIntegrationTests()
    {
        // arrange
        var tester = await _fixture.CreateSchema(_database);

        // act
        var res1 = await tester.ExecuteAsync(@"{
                actors {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        var res2 = await tester.ExecuteAsync(@"{
                actors (where : {name : { startsWith : ""Keanu"" }}) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        var res3 = await tester.ExecuteAsync(@"{
                movies {
                    items {
                        title
                    }
                }
            }");

        var res4 = await tester.ExecuteAsync(@"{
                actors(order: [{ name : ASC }]) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        var res5 = await tester.ExecuteAsync(@"{
                actors(order: [{ name : ASC }] skip: 1 take: 2) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        // assert
        await Snapshot.Create()
            .AddResult(res1, "MoviesSchema_Actors_Query")
            .AddResult(res2, "MoviesSchema_Name_StartsWith_Actors_Query")
            .AddResult(res3, "MoviesSchema_Movies_Query")
            .AddResult(res4, "MoviesSchema_Name_Desc_Sort_Actors_Query")
            .AddResult(res5, "MoviesSchema_Name_Desc_Sort_Actors_SkipOne_TakeTwo_Query")
            .MatchAsync();
    }
}
