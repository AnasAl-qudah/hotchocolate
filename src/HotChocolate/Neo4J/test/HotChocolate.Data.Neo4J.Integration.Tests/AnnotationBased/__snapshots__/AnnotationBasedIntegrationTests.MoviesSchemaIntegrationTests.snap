MoviesSchema_Actors_Query
---------------
{
  "data": {
    "actors": {
      "items": [
        {
          "name": "Carrie-Anne Moss",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Emil Eifrem",
          "actedIn": [
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Hugo Weaving",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Joel Silver",
          "actedIn": []
        },
        {
          "name": "Keanu Reeves",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Lana Wachowski",
          "actedIn": []
        },
        {
          "name": "Laurence Fishburne",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Lilly Wachowski",
          "actedIn": []
        }
      ]
    }
  }
}
---------------

MoviesSchema_Name_StartsWith_Actors_Query
---------------
{
  "data": {
    "actors": {
      "items": [
        {
          "name": "Keanu Reeves",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        }
      ]
    }
  }
}
---------------

MoviesSchema_Movies_Query
---------------
{
  "data": {
    "movies": {
      "items": [
        {
          "title": "The Matrix"
        },
        {
          "title": "The Matrix Reloaded"
        },
        {
          "title": "The Matrix Revolutions"
        }
      ]
    }
  }
}
---------------

MoviesSchema_Name_Desc_Sort_Actors_Query
---------------
{
  "data": {
    "actors": {
      "items": [
        {
          "name": "Carrie-Anne Moss",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Emil Eifrem",
          "actedIn": [
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Hugo Weaving",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Joel Silver",
          "actedIn": []
        },
        {
          "name": "Keanu Reeves",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Lana Wachowski",
          "actedIn": []
        },
        {
          "name": "Laurence Fishburne",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Lilly Wachowski",
          "actedIn": []
        }
      ]
    }
  }
}
---------------

MoviesSchema_Name_Desc_Sort_Actors_SkipOne_TakeTwo_Query
---------------
{
  "data": {
    "actors": {
      "items": [
        {
          "name": "Emil Eifrem",
          "actedIn": [
            {
              "title": "The Matrix"
            }
          ]
        },
        {
          "name": "Hugo Weaving",
          "actedIn": [
            {
              "title": "The Matrix Revolutions"
            },
            {
              "title": "The Matrix Reloaded"
            },
            {
              "title": "The Matrix"
            }
          ]
        }
      ]
    }
  }
}
---------------
